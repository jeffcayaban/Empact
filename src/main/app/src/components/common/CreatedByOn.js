import React from "react";
import CreatorName from "./CreatorName";

/**
 * Displays the argument or petition's creation details.
 */

class CreatedByOn extends React.PureComponent {

    render() {
        const { content, date } = this.props;
        return (
            <span>
                {`Created by `}
                <CreatorName content={content} />
                {` on ${date}`}
            </span>
        );
    }
}

export default CreatedByOn;