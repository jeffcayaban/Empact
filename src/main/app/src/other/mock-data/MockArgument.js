import {CRITICAL_QUESTION_IDS, EXPERT_OPINION, ROLE_ADMIN} from "../../constants";

export default {
    id: 'testId',
    isSupporting: true,
    rootPetitionId: 'testRootPetitionId',
    parentArgumentId: 'testParentArgumentId',
    subCriticalQuestionId: null,
    agreesWithCQ: true,
    criticalQuestionId: CRITICAL_QUESTION_IDS[0],
    creationDateTime: 1551716401541,
    lastUpdatedDateTime: 1551716402000,
    createdBy: {
        id: 'testId',
        username: 'testUsername',
        firstName: 'testFirstName',
        lastName: 'testLastName',
        authorities: [ROLE_ADMIN]
    },
    argumentType: EXPERT_OPINION,
    isAnonymous: false,
    sources: [],
    expert: "testExpert",
    expertDomain: 'testDomain',
    expertAssertion: 'testAssertion'
};