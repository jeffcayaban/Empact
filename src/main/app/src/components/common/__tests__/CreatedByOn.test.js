import React from 'react';
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";

import MockArgument from '../../../other/mock-data/MockArgument';
import CreatedByOn from "../CreatedByOn";

describe('CreatedByOn', () => {

    it('should render successfully', () => {
        const component = shallow(<CreatedByOn
            content={MockArgument}
            date={"testDate"}
        />);

        expect(toJson(component)).toMatchSnapshot();
    });

});
