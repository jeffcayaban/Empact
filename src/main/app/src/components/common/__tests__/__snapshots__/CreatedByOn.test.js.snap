// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatedByOn should render successfully 1`] = `
<span>
  Created by 
  <CreatorName
    content={
      Object {
        "agreesWithCQ": true,
        "argumentType": "ExpertOpinion",
        "createdBy": Object {
          "authorities": Array [
            "ROLE_ADMIN",
          ],
          "firstName": "testFirstName",
          "id": "testId",
          "lastName": "testLastName",
          "username": "testUsername",
        },
        "creationDateTime": 1551716401541,
        "criticalQuestionId": "CQ1",
        "expert": "testExpert",
        "expertAssertion": "testAssertion",
        "expertDomain": "testDomain",
        "id": "testId",
        "isAnonymous": false,
        "isSupporting": true,
        "lastUpdatedDateTime": 1551716402000,
        "parentArgumentId": "testParentArgumentId",
        "rootPetitionId": "testRootPetitionId",
        "sources": Array [],
        "subCriticalQuestionId": null,
      }
    }
  />
   on testDate
</span>
`;
